posixlib/src/main/resources/scala-native/sys/stat.c

```diff
@@ -21,9 +21,9 @@ struct scalanative_stat {
                                       length in bytes. For a typed memory object,
                                       the length in bytes. For other file types,
                                       the use of this field is unspecified. */
-    scalanative_time_t _st_atime; /** Time of last access. */
-    scalanative_time_t _st_mtime; /** Time of last data modification. */
-    scalanative_time_t _st_ctime; /** Time of last status change. */
+    scalanative_timespec st_atim; /** Time of last access. */
```

Comment:
> I do not want to make things so difficult that nothing ever gets done.
> I suspect that it is possible to introduce this PR in a way which causes
> less breakage to existing users.

Hmm, how can we introduce public api changes in a less breaking way?
Is there way better than deprecation warning?

Reply:
SN 0.5.0 is advertised/envisioned as requiring a re-build from sources and possibly requiring source changes.
That gives us some degrees of freedom to do things
a 'right' way.

So;

1) This PR should be marked in base topic as "0.5.0"
2) There should be an entry in `docs/changelog/0.5.0.md"
     briefly describing the required source change and
    pointing to an entry in javalib.rst
3) There should be a section in `docs/lib/javalib.rst'
     giving a fuller description and examples of the
     required source change (i.e. change foo._8 to 
     foo.a_time to use seconds, foo.a_tim to use timespec).

Once the dust settles on this PR, I can either help you 
set up the Sphinx environment on your system to build .rst (use "Makefile html" not 'scripts/makedoc") and .md files or do a followup PR for you to review.

Reply:
`stat.c` `scalanative_stat_init()` is exactly the kind of 
glue code I have been trying to expunge from Scala Native.
After a bunch of network code, I had hoped to get to this one before it changed.   

I think this section of code is correct/OK and will try
to queue up a follow on PR to optimize (probably to
optimize for Linux passthru with appropriate _Static guards.)
`stat` is supposed to be fast (but must always be correct).

Reply:
> There should be an entry in `docs/changelog/0.5.0.md" briefly describing the required source change and pointing to an entry in javalib.rst

> There should be a section in `docs/lib/javalib.rst' giving a fuller description and examples of the required source change

Though I changed java File internal, it doesn't leak posixlib implementation for users.

In terms of source change, I think you mean `docs/lib/posixlib`, don't you?

Of course, we should mention that java File metadata will support nanosecond resolution in the future(probably in 0.5.x?).

Reply:
I added docs here
https://github.com/scala-native/scala-native/pull/3049/commits/174e9df80eeef4fe4865eb4c78233677d529c79e
https://github.com/scala-native/scala-native/pull/3049/commits/4e2096fab3741f4509ba54f253aac34d1b6843fa